plugins/core/qPCV/PCVCommand.cpp

```diff
@@ -0,0 +1,101 @@
+#include "PCVCommand.h"
+#include "PCV.h"
+#include "qPCV.h"
+
+//qCC_db
+#include <ccGenericMesh.h>
+#include <ccHObjectCaster.h>
+#include <ccPointCloud.h>
+#include <ccProgressDialog.h>
+
+static const char COMMAND_PCV[] = "PCV";
+static const char COMMAND_PCV_N_RAYS[] = "N_RAYS";
+static const char COMMAND_PCV_IS_CLOSED[] = "IS_CLOSED";
+static const char COMMAND_PCV_180[] = "180";
+static const char COMMAND_PCV_RESOLUTION[] = "RESOLUTION";
+
+PCVCommand::PCVCommand() :
+	Command("PCV", COMMAND_PCV)
+{
+}
+
+bool PCVCommand::process(ccCommandLineInterface& cmd)
+{
+	cmd.print("[PCV]");
+
+	// Initialize to match PCV::Launch defaults
+	unsigned nRays = 256;
+	bool meshIsClosed = false;
+	bool mode360 = true;
+	unsigned resolution = 1024;
+
+	while (!cmd.arguments().empty())
+	{
+		QString arg = cmd.arguments().front();
```

Comment:
This should be `const` and you should use [takeFirst](https://doc.qt.io/qt-5/qlist.html#takeFirst):

```cpp
const QString arg = cmd.arguments().takeFirst();
```

Then remove the calls to `pop_front()` below.

Reply:
We don't want to remove the argument from the list until we've verified it's an option of PCV. Otherwise, we'll eat up the next command. I think we should leave this as is, unless you think we should take the argument out of the list and then put it back in if it doesn't match any of the PCV options.

Reply:
I'm double-checking this now and will add the const.

Reply:
Addressed in 3355f5c188c0084476d9daff63504bd7214c527c

Reply:
Ah - you're absolutely correct. I misunderstood the way this is implemented & thought it was only receiving its own commands.

Based on that - one last thing from me :-)

```cpp
const QString arg = cmd.arguments().front();
```
should actually use a reference so it isn't copying each string:
```cpp
const QString& arg = cmd.arguments().front();
```

Reply:
Addressed in 2838528cc720969a1e553c04f46ee471cd44a2cd